src/xtgeo/grid3d/grid_properties.py

```diff
@@ -34,42 +116,190 @@
+
+        dataframe_dict[prop.name] = vector
+
+    return pd.DataFrame.from_dict(dataframe_dict)
+
+
 class GridProperties(_Grid3D):
     """Class for a collection of 3D grid props, belonging to the same grid topology.
 
+     It is a thin wrapper on a list that 1) checks that the GridProperties
+     belong to the same Grid (loosely). 2) Contains operations that can be
+     called on lists of GridProperty objects for easy discoverability.
+
+     Examples::
+         >>> import xtgeo
+         >>> # Create an
+         >>> grid_properties = xtgeo.GridProperties(props=[])
+         >>> # Get the dataframe via the gridproperties object
+         >>> grid_properties.get_dataframe()
+         Empty DataFrame...
+         >>> # Convert the gridproperties to a list
+         >>> grid_properties_list = list(grid_properties)
+         >>> # Get the dataframe of the list:
+         >>> gridproperties_dataframe(grid_properties_list)
+         Empty DataFrame...
+
+    Args:
+        ncol: Deprecated argument.
+        nrow: Deprecated argument.
+        nlay: Deprecated argument.
+        props: The list of GridProperty objects.
+
     See Also:
         The :class:`GridProperty` class.
     """
 
     def __init__(
         self,
-        ncol: Optional[int] = 4,
-        nrow: Optional[int] = 3,
-        nlay: Optional[int] = 5,
+        ncol: Optional[int] = None,
+        nrow: Optional[int] = None,
+        nlay: Optional[int] = None,
+        props: List[GridProperty] = None,
     ):
-        """Instantiating.
+        dims_given = False
+        if ncol is not None:
+            warnings.warn(
+                "Initializing GridProperties with ncol is deprecated.",
+                DeprecationWarning,
+            )
+            dims_given = True
+        else:
+            ncol = 4
+        if nrow is not None:
+            warnings.warn(
+                "Initializing GridProperties with nrow is deprecated.",
+                DeprecationWarning,
+            )
+            dims_given = True
+        else:
+            nrow = 3
+        if nlay is not None:
+            warnings.warn(
+                "Initializing GridProperties with nrow is deprecated.",
+                DeprecationWarning,
+            )
+            dims_given = True
+        else:
+            nrow = 5
 
-        Args:
-            ncol: Number of columns
-            nrow: Number of rows
-            nlay: Number of layers
+        if props:
+            if dims_given:
+                raise ValueError(
+                    "Giving both ncol/nrow/nlay and props list is not supported. "
+                    "Please give just props as ncol/nrow/nlay is deprecated."
+                )
+            ncol, nrow, nlay = props[0].dimensions
 
-        """
-        super().__init__()
+        super().__init__(ncol, nrow, nlay)
 
-        self._ncol = ncol
-        self._nrow = nrow
-        self._nlay = nlay
+        self._props = props or []
```

Comment:
Consider setting `props` (not `_props`) ? It would 1) perform a consistency-check, and 2) set `names` automatically

Reply:
Done